### DISCUSSION TITLE
requesting permission on extension after gets installed in manifest version 3

### DISCUSSION THREAD

**Discussion by Alireza01** [mdn/mdn-community#394] [Platform]
Hello guys,
I have a question about requesting permission on extension when after installed in manifest version 3.

I have an extension that I want it to work only on a specific domain: "\*://www.twitch.tv/\*".

So in Firefox on manifest version 2, Easily you can put the domain in content_scripts.matches and my extension work only on the domain that I talked about. But in manifest v3, Firefox read this domain as an Optional permission and don't give the permission automatically.

![image](https://github.com/mdn/mdn-community/assets/81892736/24054eaa-da50-456e-afc3-4e49f45eac55)

I was thinking about that when my extension gets installed, I request for permission but it looks like request permission confirm dialog only opens when user clicks something. 

To be honest with you I don't think this thing that user should click something so I can request permission with confirm dialog is not necessary. The confirm dialog is enough. But here is a thing. I tested something with this extension: 
https://github.com/mdn/webextensions-examples/tree/main/dnr-dynamic-with-options

There is a security hole in Firefox that extensions can spam request permission confirm dialog even when user clicks on deny in confirm dialog. Watch this gif:

![Untitled Project](https://github.com/mdn/mdn-community/assets/81892736/d66df317-d1a4-4cab-a4db-621e16677a49)

In my opinion, after user denies this request, it should put something in "Permissions" tab in the extension settings that user denied this permission request or at be a design like the design of requesting permission in android like in first time and couple of times.

Also after I give permission to my domain it doesn't show anything in "Permissions" tab in the extension settings that I gave this permission to the extension.

I want to request that if you can remove this thing that user should click something so I can request permission with confirm dialog after my extension gets installed and fix the other things that I talked about.

Cheers

**Comment by bsmth** [comment]
Hi there! I'm very sorry you didn't get a response on this, the reason is that the GitHub Discussions here are about content on MDN Web Docs, so it's likely nobody had an answer for you, and it fell through the cracks. I would suggest you have a look here for some resources: https://github.com/mdn/webextensions-examples/tree/main?tab=readme-ov-file#problems there are a few communication channels there that might be useful for you.

I'm going to close this but feel free to reply if you think I made a mistake or if you need more help! Thanks a lot!
